src/o365/spo/commands/folder/folder-copy.ts

```diff
@@ -0,0 +1,319 @@
+          } else {
+            cmd.log(`Check #${pollCount}. JobState: ${resp.JobState}`);
+          }
+        }
+
+        for(const item of resp.Logs) {
+          const log = JSON.parse(item);
+          
+          // reject if progress error 
+          if(log.Event === "JobError" || log.Event === "JobFatalError") {
+            return reject(log.Message);
+          }
+        }
+
+        // three possible scenarios
+        // job done = success promise returned
+        // job in progress = recursive call using setTimeout returned
+        // max poll attempts flag rised = reject promise returned
+        if (resp.JobState === 0) {
+          // job done
+          resolve();
+        }
+        else if (pollCount < opts.progressMaxPollAttempts) {
+          // if the condition isn't met but the timeout hasn't elapsed, go again
+          setTimeout(checkCondition, opts.progressPollInterval, resolve, reject);
+        }
+        else {
+          reject(new Error('getCopyJobProgress timed out'));
+        }
+      }, 
+      (error: any) => {
+        retryAttemptsCount++;
+        
+        // let's retry x times in row before we give up since
+        // this is progress check and even if rejects a promise
+        // the actual copy process can success.
+        if(retryAttemptsCount <= opts.progressRetryAttempts) {
+          setTimeout(checkCondition, opts.progressPollInterval, resolve, reject);
+        }else{
+          reject(error);
+        }
+      });
+    }
+    return new Promise<any>(checkCondition);
+  }
+
+  /**
+   * Combines base and relative url 
+   * considering any missing slashes
+   * @param baseUrl https://contoso.com
+   * @param relativeUrl sites/abc
+   */
+  public urlCombine(baseUrl: string, relativeUrl: string): string {
+
+    // remove last '/' of base if exists
+    if (baseUrl.lastIndexOf('/') === baseUrl.length - 1) {
+      baseUrl = baseUrl.substring(0, baseUrl.length - 1);
+    }
+
+    // remove '/' at 0
+    if (relativeUrl.charAt(0) === '/') {
+      relativeUrl = relativeUrl.substring(1, relativeUrl.length);
+    }
+
+    // remove last '/' of next if exists
+    if (relativeUrl.lastIndexOf('/') === relativeUrl.length - 1) {
+      relativeUrl = relativeUrl.substring(0, relativeUrl.length - 1);
+    }
+
+    return `${baseUrl}/${relativeUrl}`;
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-w, --webUrl <webUrl>',
+        description: 'The URL of the site where the folder is located'
+      },
+      {
+        option: '-u, --sourceUrl <sourceUrl>',
+        description: 'Site-relative URL of the folder to copy'
+      },
+      {
+        option: '-t, --targetUrl <targetUrl>',
+        description: 'Server-relative URL where to copy the folder'
+      },
+      {
+        option: '-d, --deleteIfAlreadyExists [deleteIfAlreadyExists]',
```

Comment:
This option doesn't seem to be used. Is the implementation missing or should the option be removed?